app/controllers/manage_cmr/order_policies_controller.rb

```diff
@@ -65,20 +86,25 @@ def destroy
   end
 
   def test_endpoint_connection
-    response = echo_client.test_endpoint_connection(token, current_provider_guid)
+    response = if use_legacy_order_service?
+                 echo_client.test_endpoint_connection(token, current_provider_guid)
+               else
+                 cmr_client.test_endpoint_connection(token, current_user.provider_id)
+               end
 
-    message = if response.error?
-                response.error_message
-              else
-                'Test endpoint connection was successful.'
-              end
+    if response.error?
```

Comment:
I'm not sure if this is a sufficient way to check for errors coming from a graphql endpoint. GraphQL will always return a 200 if the request makes it to the server. You have to check the body of the response for an `errors` array to know if anything went wrong.

Reply:
I think that is covers: 
def error?
      status >= 400 || (body.is_a?(Hash) && !body['errors'].blank?)
    end

Reply:
Ah I wasn't sure where `.error?` was defined. You're right, that does look good